File: Hashing/java/allUnique.java
import java.util.HashSet;
import java.util.List;
import java.util.Set;

class AllUnique {
  public static <T> boolean allUnique(List<T> items) {
    Set<T> seen = new HashSet<>();
    for (T item : items) {
      if (!seen.add(item)) {
        return false;
      }
    }
    return true;
  }

  public static void run() {
    // Intentionally left empty.
  }
}

File: Hashing/python/allUnique.py
from typing import Iterable, TypeVar, Hashable

T = TypeVar("T", bound=Hashable)


def all_unique(items: Iterable[T]) -> bool:
    seen = set()
    for item in items:
        if item in seen:
            return False
        seen.add(item)
    return True
